core/domain/story_domain.py

```diff
@@ -1280,8 +1319,21 @@ def update_node_thumbnail_filename(self, node_id, new_thumbnail_filename):
         if node_index is None:
             raise ValueError(
                 'The node with id %s is not part of this story' % node_id)
-        self.story_contents.nodes[node_index].thumbnail_filename = (
-            new_thumbnail_filename)
+        file_system_class = fs_services.get_entity_file_system_class()
+        fs = fs_domain.AbstractFileSystem(file_system_class(
+            feconf.ENTITY_TYPE_STORY, self.id))
+
+        filepath = '%s/%s' % (
+            constants.ASSET_TYPE_THUMBNAIL, new_thumbnail_filename)
+        if fs.isfile(filepath):
+            self.story_contents.nodes[node_index].thumbnail_filename = (
+                new_thumbnail_filename)
+            self.story_contents.nodes[node_index].thumbnail_size_in_bytes = (
+                len(fs.get(filepath)))
+        else:
+            raise Exception(
```

Comment:
Ditto here. Need to add tests for this to have the backend coverage tests pass.

Reply:
done